examples/cube/main.rs

```diff
@@ -167,16 +188,80 @@ impl<R: gfx::Resources> gfx_app::Application<R> for App<R> {
         encoder.clear(&self.bundle.data.out_color, [0.1, 0.2, 0.3, 1.0]);
         encoder.clear_depth(&self.bundle.data.out_depth, 1.0);
         self.bundle.encode(encoder);
+
+        if self.screenshot == Screenshot::Ready {
+            use gfx::memory::Typed;
+
+            let (w, h, _, _) = self.bundle.data.out_color.get_dimensions();
+            encoder.copy_texture_to_buffer_raw(
+                self.bundle.data.out_color.raw().get_texture(),
+                None,
+                gfx::texture::RawImageInfo {
+                    xoffset: 0,
+                    yoffset: 0,
+                    zoffset: 0,
+                    width: w,
+                    height: h,
+                    depth: 0,
+                    format: ColorFormat::get_format(),
+                    mipmap: 0,
+                },
+                self.download.raw(),
+                0
+            ).unwrap();
+	          println!("screenshot taken");
+            self.screenshot = Screenshot::Some;
+        }
     }
 
-    fn on_resize(&mut self, window_targets: gfx_app::WindowTargets<R>) {
+    fn on_resize_ext<F: gfx::Factory<R>>(&mut self,
+                                         factory: &mut F,
+                                         window_targets: gfx_app::WindowTargets<R>) {
+        use gfx::traits::FactoryExt;
+
+        if self.screenshot == Screenshot::Some {
+            let (w, h, _, _) = self.bundle.data.out_depth.get_dimensions();
+            let reader = factory.read_mapping(&self.download).unwrap();
+            let mut data = Vec::with_capacity(w as usize * h as usize * 4);
+            for l in reader.chunks(w as usize).rev() {
+                for &v in l {
+                    data.push(v[0]);
+                    data.push(v[1]);
+                    data.push(v[2]);
+                }
+            }
+	          let path = "screen.png";
+            image::save_buffer(path,
+                               &data,
+                               w as u32,
+                               h as u32,
+                               image::ColorType::RGB(8)).unwrap();
+            println!("screenshot saved to {}", path);
+            self.screenshot = Screenshot::None;
+        }
+
         self.bundle.data.out_color = window_targets.color;
         self.bundle.data.out_depth = window_targets.depth;
+        let (w, h, _, _) = self.bundle.data.out_color.get_dimensions();
+        self.download = factory.create_download_buffer(w as usize * h as usize)
+            .unwrap();
 
         // In this example the transform is static except for window resizes.
         let proj = cgmath::perspective(cgmath::deg(45.0f32), window_targets.aspect_ratio, 1.0, 10.0);
         self.bundle.data.transform = (proj * default_view().mat).into();
     }
+
+    fn on(&mut self, event: winit::Event) {
+        use winit::Event::*;
```

Comment:
nit: lets not do `*` and instead just do `use winit::{Event, ElementState, VirtualKeyCode as Key};`